### DISCUSSION TITLE
Mac (M3 chip) problems linking ncurses for menuconfig

### DISCUSSION THREAD

**Discussion by NageshEranki** [bitcraze/discussions#1235] [Q&A] [answered]
Hi

I'm trying to use 'make menuconfig' to customize my build. But am running into the following error:

<img width="760" alt="Screenshot 2024-03-17 at 7 14 29 PM" src="https://github.com/bitcraze/discussions/assets/28665953/fa5cf6be-04fc-44f0-8353-a0f0dba26053">

I ran ```brew install ncurses``` on a Rosetta terminal:

<img width="1446" alt="Screenshot 2024-03-17 at 7 16 26 PM" src="https://github.com/bitcraze/discussions/assets/28665953/f21d62f7-2aaf-4674-844b-31e287bbdca2">

I still see the same : *** Unable to find the ncurses libraries or the required header files.*** error

So I tried modifying the ```check-lxdialog.h``` file to manually link to the ```ncurses.h``` file brew installed it in the above step:

<img width="873" alt="Screenshot 2024-03-17 at 7 22 07 PM" src="https://github.com/bitcraze/discussions/assets/28665953/e15179d1-0817-4e17-8fec-379d36f2ee66">

Sadly, this also didn't work :( 

Nagesh

**Comment by ataffanel** [comment, marked as answer]
Unfortunatly, it works for me on our dev M1 mac and I am not sure what we did more than you to get it to work. At least it shows that it can work.

One clue I found while searching for the problem, is that you can also do ```make nconfig``` which also work for me but do not require ncurses so it might work for you as well.

```gconfig``` and ```xconfig``` might also work but requires even bigger library so they would not be my first try.

**Comment by NageshEranki** [comment]
```make nconfig``` works! Thank you so much <3

Note: Restored the ```check-lxdialog.h``` file to the original and ran ```make nconfig``` in root of ```crazyflie-firmware```.

**Comment by ignabelitzky** [comment]
Did you try this:
```Bash
brew install ncurses-devel
```

Install the devel package on Linux works for me everytime I'm in a new system. Please try that and let me know.

**Reply by qiaopengjun5162** [reply]
brew install ncurses-devel
Running `brew update --auto-update`...
Warning: No available formula with the name "ncurses-devel". Did you mean ncurses?
==> Searching for similarly named formulae and casks...
==> Formulae
ncurses ✔

To install ncurses ✔, run:
  brew install ncurses ✔
